// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: execs.proto

package grpcapipb

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on ExecLoginRequest with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *ExecLoginRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ExecLoginRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ExecLoginRequestMultiError, or nil if none found.
func (m *ExecLoginRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *ExecLoginRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Username

	// no validation rules for Password

	if len(errors) > 0 {
		return ExecLoginRequestMultiError(errors)
	}

	return nil
}

// ExecLoginRequestMultiError is an error wrapping multiple validation errors
// returned by ExecLoginRequest.ValidateAll() if the designated constraints
// aren't met.
type ExecLoginRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ExecLoginRequestMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ExecLoginRequestMultiError) AllErrors() []error { return m }

// ExecLoginRequestValidationError is the validation error returned by
// ExecLoginRequest.Validate if the designated constraints aren't met.
type ExecLoginRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ExecLoginRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ExecLoginRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ExecLoginRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ExecLoginRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ExecLoginRequestValidationError) ErrorName() string { return "ExecLoginRequestValidationError" }

// Error satisfies the builtin error interface
func (e ExecLoginRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sExecLoginRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ExecLoginRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ExecLoginRequestValidationError{}

// Validate checks the field values on ExecLoginResponse with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *ExecLoginResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ExecLoginResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ExecLoginResponseMultiError, or nil if none found.
func (m *ExecLoginResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *ExecLoginResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Status

	// no validation rules for Token

	if len(errors) > 0 {
		return ExecLoginResponseMultiError(errors)
	}

	return nil
}

// ExecLoginResponseMultiError is an error wrapping multiple validation errors
// returned by ExecLoginResponse.ValidateAll() if the designated constraints
// aren't met.
type ExecLoginResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ExecLoginResponseMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ExecLoginResponseMultiError) AllErrors() []error { return m }

// ExecLoginResponseValidationError is the validation error returned by
// ExecLoginResponse.Validate if the designated constraints aren't met.
type ExecLoginResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ExecLoginResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ExecLoginResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ExecLoginResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ExecLoginResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ExecLoginResponseValidationError) ErrorName() string {
	return "ExecLoginResponseValidationError"
}

// Error satisfies the builtin error interface
func (e ExecLoginResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sExecLoginResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ExecLoginResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ExecLoginResponseValidationError{}

// Validate checks the field values on ForgotPasswordResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *ForgotPasswordResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ForgotPasswordResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ForgotPasswordResponseMultiError, or nil if none found.
func (m *ForgotPasswordResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *ForgotPasswordResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Confirmation

	// no validation rules for Message

	if len(errors) > 0 {
		return ForgotPasswordResponseMultiError(errors)
	}

	return nil
}

// ForgotPasswordResponseMultiError is an error wrapping multiple validation
// errors returned by ForgotPasswordResponse.ValidateAll() if the designated
// constraints aren't met.
type ForgotPasswordResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ForgotPasswordResponseMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ForgotPasswordResponseMultiError) AllErrors() []error { return m }

// ForgotPasswordResponseValidationError is the validation error returned by
// ForgotPasswordResponse.Validate if the designated constraints aren't met.
type ForgotPasswordResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ForgotPasswordResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ForgotPasswordResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ForgotPasswordResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ForgotPasswordResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ForgotPasswordResponseValidationError) ErrorName() string {
	return "ForgotPasswordResponseValidationError"
}

// Error satisfies the builtin error interface
func (e ForgotPasswordResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sForgotPasswordResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ForgotPasswordResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ForgotPasswordResponseValidationError{}

// Validate checks the field values on ForgotPasswordRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *ForgotPasswordRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ForgotPasswordRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ForgotPasswordRequestMultiError, or nil if none found.
func (m *ForgotPasswordRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *ForgotPasswordRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Email

	if len(errors) > 0 {
		return ForgotPasswordRequestMultiError(errors)
	}

	return nil
}

// ForgotPasswordRequestMultiError is an error wrapping multiple validation
// errors returned by ForgotPasswordRequest.ValidateAll() if the designated
// constraints aren't met.
type ForgotPasswordRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ForgotPasswordRequestMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ForgotPasswordRequestMultiError) AllErrors() []error { return m }

// ForgotPasswordRequestValidationError is the validation error returned by
// ForgotPasswordRequest.Validate if the designated constraints aren't met.
type ForgotPasswordRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ForgotPasswordRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ForgotPasswordRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ForgotPasswordRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ForgotPasswordRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ForgotPasswordRequestValidationError) ErrorName() string {
	return "ForgotPasswordRequestValidationError"
}

// Error satisfies the builtin error interface
func (e ForgotPasswordRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sForgotPasswordRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ForgotPasswordRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ForgotPasswordRequestValidationError{}

// Validate checks the field values on Confimation with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *Confimation) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on Confimation with the rules defined in
// the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in ConfimationMultiError, or
// nil if none found.
func (m *Confimation) ValidateAll() error {
	return m.validate(true)
}

func (m *Confimation) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Confirmation

	if len(errors) > 0 {
		return ConfimationMultiError(errors)
	}

	return nil
}

// ConfimationMultiError is an error wrapping multiple validation errors
// returned by Confimation.ValidateAll() if the designated constraints aren't met.
type ConfimationMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ConfimationMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ConfimationMultiError) AllErrors() []error { return m }

// ConfimationValidationError is the validation error returned by
// Confimation.Validate if the designated constraints aren't met.
type ConfimationValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ConfimationValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ConfimationValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ConfimationValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ConfimationValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ConfimationValidationError) ErrorName() string { return "ConfimationValidationError" }

// Error satisfies the builtin error interface
func (e ConfimationValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sConfimation.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ConfimationValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ConfimationValidationError{}

// Validate checks the field values on ResetPasswordRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *ResetPasswordRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ResetPasswordRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ResetPasswordRequestMultiError, or nil if none found.
func (m *ResetPasswordRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *ResetPasswordRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for ResetCode

	// no validation rules for NewPassword

	// no validation rules for ConfirmPassword

	if len(errors) > 0 {
		return ResetPasswordRequestMultiError(errors)
	}

	return nil
}

// ResetPasswordRequestMultiError is an error wrapping multiple validation
// errors returned by ResetPasswordRequest.ValidateAll() if the designated
// constraints aren't met.
type ResetPasswordRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ResetPasswordRequestMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ResetPasswordRequestMultiError) AllErrors() []error { return m }

// ResetPasswordRequestValidationError is the validation error returned by
// ResetPasswordRequest.Validate if the designated constraints aren't met.
type ResetPasswordRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ResetPasswordRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ResetPasswordRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ResetPasswordRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ResetPasswordRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ResetPasswordRequestValidationError) ErrorName() string {
	return "ResetPasswordRequestValidationError"
}

// Error satisfies the builtin error interface
func (e ResetPasswordRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sResetPasswordRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ResetPasswordRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ResetPasswordRequestValidationError{}

// Validate checks the field values on UpdatePasswordResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *UpdatePasswordResponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on UpdatePasswordResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// UpdatePasswordResponseMultiError, or nil if none found.
func (m *UpdatePasswordResponse) ValidateAll() error {
	return m.validate(true)
}

func (m *UpdatePasswordResponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for PasswordUpdated

	// no validation rules for Token

	if len(errors) > 0 {
		return UpdatePasswordResponseMultiError(errors)
	}

	return nil
}

// UpdatePasswordResponseMultiError is an error wrapping multiple validation
// errors returned by UpdatePasswordResponse.ValidateAll() if the designated
// constraints aren't met.
type UpdatePasswordResponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m UpdatePasswordResponseMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m UpdatePasswordResponseMultiError) AllErrors() []error { return m }

// UpdatePasswordResponseValidationError is the validation error returned by
// UpdatePasswordResponse.Validate if the designated constraints aren't met.
type UpdatePasswordResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e UpdatePasswordResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e UpdatePasswordResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e UpdatePasswordResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e UpdatePasswordResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e UpdatePasswordResponseValidationError) ErrorName() string {
	return "UpdatePasswordResponseValidationError"
}

// Error satisfies the builtin error interface
func (e UpdatePasswordResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sUpdatePasswordResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = UpdatePasswordResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = UpdatePasswordResponseValidationError{}

// Validate checks the field values on UpdatePasswordRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *UpdatePasswordRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on UpdatePasswordRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// UpdatePasswordRequestMultiError, or nil if none found.
func (m *UpdatePasswordRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *UpdatePasswordRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	// no validation rules for CurrentPassword

	// no validation rules for NewPassword

	if len(errors) > 0 {
		return UpdatePasswordRequestMultiError(errors)
	}

	return nil
}

// UpdatePasswordRequestMultiError is an error wrapping multiple validation
// errors returned by UpdatePasswordRequest.ValidateAll() if the designated
// constraints aren't met.
type UpdatePasswordRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m UpdatePasswordRequestMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m UpdatePasswordRequestMultiError) AllErrors() []error { return m }

// UpdatePasswordRequestValidationError is the validation error returned by
// UpdatePasswordRequest.Validate if the designated constraints aren't met.
type UpdatePasswordRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e UpdatePasswordRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e UpdatePasswordRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e UpdatePasswordRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e UpdatePasswordRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e UpdatePasswordRequestValidationError) ErrorName() string {
	return "UpdatePasswordRequestValidationError"
}

// Error satisfies the builtin error interface
func (e UpdatePasswordRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sUpdatePasswordRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = UpdatePasswordRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = UpdatePasswordRequestValidationError{}

// Validate checks the field values on EmptyRequest with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *EmptyRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on EmptyRequest with the rules defined
// in the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in EmptyRequestMultiError, or
// nil if none found.
func (m *EmptyRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *EmptyRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return EmptyRequestMultiError(errors)
	}

	return nil
}

// EmptyRequestMultiError is an error wrapping multiple validation errors
// returned by EmptyRequest.ValidateAll() if the designated constraints aren't met.
type EmptyRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m EmptyRequestMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m EmptyRequestMultiError) AllErrors() []error { return m }

// EmptyRequestValidationError is the validation error returned by
// EmptyRequest.Validate if the designated constraints aren't met.
type EmptyRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e EmptyRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e EmptyRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e EmptyRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e EmptyRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e EmptyRequestValidationError) ErrorName() string { return "EmptyRequestValidationError" }

// Error satisfies the builtin error interface
func (e EmptyRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sEmptyRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = EmptyRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = EmptyRequestValidationError{}

// Validate checks the field values on ExecLogoutResponsesponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *ExecLogoutResponsesponse) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ExecLogoutResponsesponse with the
// rules defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// ExecLogoutResponsesponseMultiError, or nil if none found.
func (m *ExecLogoutResponsesponse) ValidateAll() error {
	return m.validate(true)
}

func (m *ExecLogoutResponsesponse) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for LoggedOut

	if len(errors) > 0 {
		return ExecLogoutResponsesponseMultiError(errors)
	}

	return nil
}

// ExecLogoutResponsesponseMultiError is an error wrapping multiple validation
// errors returned by ExecLogoutResponsesponse.ValidateAll() if the designated
// constraints aren't met.
type ExecLogoutResponsesponseMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ExecLogoutResponsesponseMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ExecLogoutResponsesponseMultiError) AllErrors() []error { return m }

// ExecLogoutResponsesponseValidationError is the validation error returned by
// ExecLogoutResponsesponse.Validate if the designated constraints aren't met.
type ExecLogoutResponsesponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ExecLogoutResponsesponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ExecLogoutResponsesponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ExecLogoutResponsesponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ExecLogoutResponsesponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ExecLogoutResponsesponseValidationError) ErrorName() string {
	return "ExecLogoutResponsesponseValidationError"
}

// Error satisfies the builtin error interface
func (e ExecLogoutResponsesponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sExecLogoutResponsesponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ExecLogoutResponsesponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ExecLogoutResponsesponseValidationError{}

// Validate checks the field values on DeleteExecsConfirmation with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *DeleteExecsConfirmation) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on DeleteExecsConfirmation with the
// rules defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// DeleteExecsConfirmationMultiError, or nil if none found.
func (m *DeleteExecsConfirmation) ValidateAll() error {
	return m.validate(true)
}

func (m *DeleteExecsConfirmation) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Status

	if len(errors) > 0 {
		return DeleteExecsConfirmationMultiError(errors)
	}

	return nil
}

// DeleteExecsConfirmationMultiError is an error wrapping multiple validation
// errors returned by DeleteExecsConfirmation.ValidateAll() if the designated
// constraints aren't met.
type DeleteExecsConfirmationMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m DeleteExecsConfirmationMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m DeleteExecsConfirmationMultiError) AllErrors() []error { return m }

// DeleteExecsConfirmationValidationError is the validation error returned by
// DeleteExecsConfirmation.Validate if the designated constraints aren't met.
type DeleteExecsConfirmationValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e DeleteExecsConfirmationValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e DeleteExecsConfirmationValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e DeleteExecsConfirmationValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e DeleteExecsConfirmationValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e DeleteExecsConfirmationValidationError) ErrorName() string {
	return "DeleteExecsConfirmationValidationError"
}

// Error satisfies the builtin error interface
func (e DeleteExecsConfirmationValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sDeleteExecsConfirmation.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = DeleteExecsConfirmationValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = DeleteExecsConfirmationValidationError{}

// Validate checks the field values on ExecId with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *ExecId) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ExecId with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in ExecIdMultiError, or nil if none found.
func (m *ExecId) ValidateAll() error {
	return m.validate(true)
}

func (m *ExecId) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	if len(errors) > 0 {
		return ExecIdMultiError(errors)
	}

	return nil
}

// ExecIdMultiError is an error wrapping multiple validation errors returned by
// ExecId.ValidateAll() if the designated constraints aren't met.
type ExecIdMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ExecIdMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ExecIdMultiError) AllErrors() []error { return m }

// ExecIdValidationError is the validation error returned by ExecId.Validate if
// the designated constraints aren't met.
type ExecIdValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ExecIdValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ExecIdValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ExecIdValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ExecIdValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ExecIdValidationError) ErrorName() string { return "ExecIdValidationError" }

// Error satisfies the builtin error interface
func (e ExecIdValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sExecId.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ExecIdValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ExecIdValidationError{}

// Validate checks the field values on ExecIds with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *ExecIds) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on ExecIds with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in ExecIdsMultiError, or nil if none found.
func (m *ExecIds) ValidateAll() error {
	return m.validate(true)
}

func (m *ExecIds) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	for idx, item := range m.GetIds() {
		_, _ = idx, item

		if all {
			switch v := interface{}(item).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, ExecIdsValidationError{
						field:  fmt.Sprintf("Ids[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, ExecIdsValidationError{
						field:  fmt.Sprintf("Ids[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return ExecIdsValidationError{
					field:  fmt.Sprintf("Ids[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	if len(errors) > 0 {
		return ExecIdsMultiError(errors)
	}

	return nil
}

// ExecIdsMultiError is an error wrapping multiple validation errors returned
// by ExecIds.ValidateAll() if the designated constraints aren't met.
type ExecIdsMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ExecIdsMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ExecIdsMultiError) AllErrors() []error { return m }

// ExecIdsValidationError is the validation error returned by ExecIds.Validate
// if the designated constraints aren't met.
type ExecIdsValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ExecIdsValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ExecIdsValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ExecIdsValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ExecIdsValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ExecIdsValidationError) ErrorName() string { return "ExecIdsValidationError" }

// Error satisfies the builtin error interface
func (e ExecIdsValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sExecIds.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ExecIdsValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ExecIdsValidationError{}

// Validate checks the field values on GetExecsRequest with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *GetExecsRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on GetExecsRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// GetExecsRequestMultiError, or nil if none found.
func (m *GetExecsRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *GetExecsRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if all {
		switch v := interface{}(m.GetExec()).(type) {
		case interface{ ValidateAll() error }:
			if err := v.ValidateAll(); err != nil {
				errors = append(errors, GetExecsRequestValidationError{
					field:  "Exec",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		case interface{ Validate() error }:
			if err := v.Validate(); err != nil {
				errors = append(errors, GetExecsRequestValidationError{
					field:  "Exec",
					reason: "embedded message failed validation",
					cause:  err,
				})
			}
		}
	} else if v, ok := interface{}(m.GetExec()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return GetExecsRequestValidationError{
				field:  "Exec",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	for idx, item := range m.GetSortBy() {
		_, _ = idx, item

		if all {
			switch v := interface{}(item).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, GetExecsRequestValidationError{
						field:  fmt.Sprintf("SortBy[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, GetExecsRequestValidationError{
						field:  fmt.Sprintf("SortBy[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return GetExecsRequestValidationError{
					field:  fmt.Sprintf("SortBy[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	// no validation rules for PageNumber

	// no validation rules for PageSize

	if len(errors) > 0 {
		return GetExecsRequestMultiError(errors)
	}

	return nil
}

// GetExecsRequestMultiError is an error wrapping multiple validation errors
// returned by GetExecsRequest.ValidateAll() if the designated constraints
// aren't met.
type GetExecsRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m GetExecsRequestMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m GetExecsRequestMultiError) AllErrors() []error { return m }

// GetExecsRequestValidationError is the validation error returned by
// GetExecsRequest.Validate if the designated constraints aren't met.
type GetExecsRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e GetExecsRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e GetExecsRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e GetExecsRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e GetExecsRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e GetExecsRequestValidationError) ErrorName() string { return "GetExecsRequestValidationError" }

// Error satisfies the builtin error interface
func (e GetExecsRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sGetExecsRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = GetExecsRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = GetExecsRequestValidationError{}

// Validate checks the field values on Exec with the rules defined in the proto
// definition for this message. If any rules are violated, the first error
// encountered is returned, or nil if there are no violations.
func (m *Exec) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on Exec with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in ExecMultiError, or nil if none found.
func (m *Exec) ValidateAll() error {
	return m.validate(true)
}

func (m *Exec) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	// no validation rules for FirstName

	// no validation rules for LastName

	// no validation rules for Email

	// no validation rules for Role

	// no validation rules for Username

	// no validation rules for Password

	// no validation rules for PasswordChangedAt

	// no validation rules for UserCreatedAt

	// no validation rules for PasswordResetToken

	// no validation rules for PasswordTokenExpires

	// no validation rules for InactiveStatus

	if len(errors) > 0 {
		return ExecMultiError(errors)
	}

	return nil
}

// ExecMultiError is an error wrapping multiple validation errors returned by
// Exec.ValidateAll() if the designated constraints aren't met.
type ExecMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ExecMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ExecMultiError) AllErrors() []error { return m }

// ExecValidationError is the validation error returned by Exec.Validate if the
// designated constraints aren't met.
type ExecValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ExecValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ExecValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ExecValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ExecValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ExecValidationError) ErrorName() string { return "ExecValidationError" }

// Error satisfies the builtin error interface
func (e ExecValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sExec.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ExecValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ExecValidationError{}

// Validate checks the field values on Execs with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *Execs) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on Execs with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in ExecsMultiError, or nil if none found.
func (m *Execs) ValidateAll() error {
	return m.validate(true)
}

func (m *Execs) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	for idx, item := range m.GetExecs() {
		_, _ = idx, item

		if all {
			switch v := interface{}(item).(type) {
			case interface{ ValidateAll() error }:
				if err := v.ValidateAll(); err != nil {
					errors = append(errors, ExecsValidationError{
						field:  fmt.Sprintf("Execs[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			case interface{ Validate() error }:
				if err := v.Validate(); err != nil {
					errors = append(errors, ExecsValidationError{
						field:  fmt.Sprintf("Execs[%v]", idx),
						reason: "embedded message failed validation",
						cause:  err,
					})
				}
			}
		} else if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return ExecsValidationError{
					field:  fmt.Sprintf("Execs[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	if len(errors) > 0 {
		return ExecsMultiError(errors)
	}

	return nil
}

// ExecsMultiError is an error wrapping multiple validation errors returned by
// Execs.ValidateAll() if the designated constraints aren't met.
type ExecsMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m ExecsMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m ExecsMultiError) AllErrors() []error { return m }

// ExecsValidationError is the validation error returned by Execs.Validate if
// the designated constraints aren't met.
type ExecsValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ExecsValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ExecsValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ExecsValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ExecsValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ExecsValidationError) ErrorName() string { return "ExecsValidationError" }

// Error satisfies the builtin error interface
func (e ExecsValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sExecs.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ExecsValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ExecsValidationError{}
